### ISSUE TITLE
request for csparse ebuild

### ISSUE THREAD

**Issue by Alessandro-Barbieri** [cschwan/sage-on-gentoo#694]
Since do are the maintainer of cxsparse, do you also have an ebuild of [csparse](https://people.sc.fsu.edu/~jburkardt/c_src/csparse/csparse.html) under the carpet 👉👈?

**Comment by kiwifb** [comment]
That's not distributed as part of suitesparse so I don't [I am currently doing a terrible job maintaining it by the way, I'll have to block some time just for it].

By the look of it, it is in a very primitive state. No package versioning or even version control. If he was working in my institution, I would have some words with that guy about distributing their software. If there had been at least some github/gitlab repo I would have packaged it. But not in this form.

**Comment by kiwifb** [comment]
Then if there is no ebuild for it now, this is an oversight.

Putting all component in one package is possible I guess, that would solve some issues. But the same problem remain, the build system is crap and the meta build more so (he says as he just did package it like that for sagemath).

**Comment by kiwifb** [comment]
Yes the original stuff from bicatali didn't include csparse and I didn't notice. I'll include it as soon as I can give it some love.

**Comment by kiwifb** [comment]
I have started to give some love to it. Next time open an issue on https://github.com/kiwifb/suitesparse this is really where it should go. In the present state I guess it is ready to produce a csparse ebuild but I want to do some stuff on cholmod before releasing the rest of it towards the main tree.

**Comment by kiwifb** [comment]
This is getting rather interesting. Do we need an ebuild for csparse? The latest ubuntu doesn't offer it. And the makefile at https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/master/Makefile is rather interesting. By default csparse is built but not installed. Which would be why it is not distributed. The next interesting things is that cxsparse is generated from csparse and csparse is just a subcase of cxsparse. If you are interested in functions in csparse, they should be available in cxsparse - possibly with a prefixed or suffixed name.

Do you have legacy code lying around that needs csparse specifically? If you do, it probably can be migrated to cxsparse.

**Comment by Alessandro-Barbieri** [comment]
I want to unbundle csparse from casadi
https://github.com/casadi/casadi/tree/master/external_packages/CSparse

**Comment by kiwifb** [comment]
OK, that's a bit ugly and proves what mess suitesparse is when people choose to just vendor the bit of it they are interested in. That certainly put some validity to your request.

All those external packages haven't been updated in years if I read github timestamp correctly. Some many time bombs :(
